app/src/main/java/com/realityexpander/tasky/core/util/ConnectivityObserver/ConnectivityObserverImpl.kt

```diff
@@ -17,6 +20,61 @@ class ConnectivityObserverImpl(
 
     // use connectivityManager.requestNetwork(networkRequest, networkCallback) for api levels lower than 24
 
+    companion object {
```

Comment:
Is everything in this class really useful tho?

Like, do we care if its GSM, Metered, Vpn, ethernet, etc?

Philipp has a video where he shows how to observe the network: https://youtu.be/TzV0oCRDNfM

Reply:
Im using his observer here. 

I was thinking I would have to support all those others but I just made a companion object with a simple bash script to ping the network for internet availability in some parts to avoid making unnecessary calls.